### ISSUE TITLE
Merge functionalities of collision_frequency, collision_rate_{ion,electron}_ion

### ISSUE THREAD

**Issue by StanczakDominik** [PlasmaPy/PlasmaPy#312]
I came across this working on #307 and #303. As discussed during the last code review session, we have three functions that sort of do the same thing, but not quite. In transport.py, there's @lemmatum's `collision_frequency` function. In [Callen's notes](http://homepages.cae.wisc.edu/~callen/chap2.pdf), it's equation 2.14.

https://github.com/PlasmaPy/PlasmaPy/blob/6d7942dc3ca473d7e9bc999d1e8a6e71ec10b8ca/plasmapy/physics/transport.py#L526-L684
And then in parameters.py there's @samurai688's version - two functions named `collision_rate_ion_ion` and `collision_rate_electron_ion`. However, Callen calls them the "fundamental ion (and electron, respectively) collison frequencies":
https://github.com/PlasmaPy/PlasmaPy/blob/6d7942dc3ca473d7e9bc999d1e8a6e71ec10b8ca/plasmapy/physics/parameters.py#L562-L688
From what I was able to gather, they're related by (in pseudocode):
```python
from conveniently_named_variables import *
collision_rate_electron_ion = 4 / (3 * np.sqrt(np.pi)) collision_frequency(thermal_speed(..., 'e' ...)
collision_rate_ion_ion = (m_e/m_i)**2 *(T_e/T_i)**1.5 * Z_i**2 * v_e / np.sqrt(2)
```
where that last square root of 2 is somehow related to reduced masses and interparticle motion in the ion-ion analysis. So... it looks like it's not going to be too trivial to simply replace these! I'm not really sure how to proceed here. I'm thinking about replacing some of the "fundamental" calculations in `rate` functions (they start from the Coloumb logarithm) in terms of our `collision_frequency` as seen in my pseudocode above.

**Comment by StanczakDominik** [comment]
Also, I'm happy I got this github syntax to work!

**Comment by samurai688** [comment]
Nice, looks right to me!  I think you've cracked the code.

It sounds like what we need is to understand and document all these different flavors of collision frequencies, which you've begun here. Then we could make the collision_frequency function able to produce all the different flavors on request, with some sort of sensible defaults. Then in each assorted function that needs collision frequency, be it hall parameter or transport or whatnot, they can call the function in the proper way for their need.

What do you think?

**Comment by StanczakDominik** [comment]
Sounds reasonable. This means we'd need to sort of bake in the ability for `collision_frequency` to calculate Maxwellian averages, as that's - as I understand it - your versions do. Maybe the sqrt(2) could even pop out algorithmically via some sort of reduced mass calculation

**Comment by samurai688** [comment]
Yeah, I feel like we would want a syntax to specify which collision
frequency we mean.  I've never really learned it all very well to be
honest.  It's like if somebody asks you what the "collision frequency" is,
you can give an order of magnitude, but if they have something specific in
mind than it's like, wait, do we mean electrons or ions, and then there's
often some factors of 4 or square roots of pi or some such.

On Sun, Mar 25, 2018 at 10:14 AM, Dominik Stańczak <notifications@github.com

**Comment by lemmatum** [comment]
Some of those factors of `sqrt(2)` may be due to different definitions of thermal velocity. I've been noticing that quite a few publications are either assuming `v_Th = sqrt(k * T / m)` or quietly switching between that and `v_Th = sqrt(2 * k * T / m)` and dropping factors of 2 along the way.
@samurai688 how does Braginskii define the thermal velocity in the publications you used to writeup the `transport.py` code?

**Comment by lemmatum** [comment]
What would make this task easier is to take each definition for the collision frequency and split it up into its constituent parts: `freq = n * sigma * V * cou_log`, where `n` is the density, `sigma` is the collisional cross section, `V` is the velocity of the incoming particle, and `cou_log` is the Coulomb logarithm accounting for small angle collisions.

**Comment by lemmatum** [comment]
So #303 is going to refactor `collision_rate_ion_ion` in terms of `freq = n * sigma * V * cou_log` and also add tests for `collision_rate_ion_ion` . From there I'll open a new PR to refactor in terms of `collision_frequency` from `collisions.py`. If that works, then we should just remove `collision_rate_ion_ion` altogether.

There are a few places to check for constant factor differences:
- [ ] reduced_mass causes mass of identical particles to map to `m/2` instead of `m`. This mainly affects `b_perp`, and could slightly affect `Coulomb_logarithm`. Though `coulomb_log_val` uses `Coulomb_logarithm` so this will be irrelevant for  `collision_rate_ion_ion`.
- [ ] different definitions of collisional frequency cause something like a factor of `np.sqrt(8 / np.pi) / 3`

We should also consider adding some functionality to `collision_frequency` and other `collisions.py` functions to use a particular mass instead of reduced mass.

**Comment by lemmatum** [comment]
So Callen's equation 2.14 is identical to my implementation of `collision_frequency, at least this portion with the exact equality is:
![image](https://user-images.githubusercontent.com/28945309/38535080-2200611c-3c3f-11e8-888f-114acf3816e9.png)


I have not idea where `collision_rate_ion_ion` and `collision_rate_electron_ion` get these strange coefficients from. (Where in the world does the factor of `3` come from?!). If equation 2.14 was indeed the aim for these functions then I would propose that we verify `collision_frequency` is using the right parameters in each collision case (checking the correct mass and density, ion or electron with respect to Callen's equation) and then remove `collision_rate_ion_ion` and `collision_rate_electron_ion`.

**Comment by lemmatum** [comment]
Here is my work for demonstrating `collision_frequency` is consistent with eq 2.14 in Callen

* Equations 1-3 are definitions I used in `collision_frequency` based on Chen. Note that subscripts 1 and 2 refer to particle species
* Eq 4 is substituting Eqs 2 & 3 into Eq 1
* Eq 5 is just some algebra
* Eq 6 assumes one of the particles is and electron and the other particle is an ion. Therefore Z_1 becomes Z_i and Z_2 goes to 1 and vanishes. This equation also assumes that mass is electron mass (which I'm 99% sure I implemented for the electron-ion case in `collision_frequency`) and usually velocity will be electron thermal velocity. Finally, it assumes that n is n_i (ion density). In `collision_frequency` this is up to the user's discretion as it just has a general density argument.
* Eq 7 assumes quasi-neutrality between ions and electrons n_e = Z_i n_i, and does some rearranging in getting it to the same form as in Callen eq 2.14

![image](https://user-images.githubusercontent.com/28945309/38535690-f77b5516-3c41-11e8-8a6d-3c05c541c108.png)
